sources/packages/api/src/services/institution-location/institution-location.service.ts

```diff
@@ -24,6 +24,20 @@ export class InstitutionLocationService extends RecordDataModelService<Instituti
   async getById(id: number): Promise<InstitutionLocation> {
     return await this.repo.findOne(id);
   }
+  
+  async getInstitutionLocationById(id: number): Promise<InstitutionLocation> {
+    return this.repo
+      .createQueryBuilder("institution_location")
+      .select([
+        "institution_location.name",
+        "institution_location.data",
+        "institution_location.id",
+        "institution",
+      ])
+      .leftJoin("institution_location.institution", "institution")
```

Comment:
It seems that there is no need to return the entire institution record here. Please consider returning only what is need for now.
I believe that you can also use the leftJoinAndSelect, if you think that it would be better.